v0/destinations/google_adwords_remarketing_lists/util.js

```diff
@@ -0,0 +1,179 @@
+        }
+      }
+    },
+    headers,
+    method
+  };
+  const response = await httpSend(jobCreatingRequest);
+  return response;
+};
+/**
+ * This function helps to put user details in a offlineUserDataJobs
+ * @param endpoint
+ * @param headers
+ * @param method
+ * @param jobId
+ * @param body
+ */
+
+const addUserToJob = async (endpoint, headers, method, jobId, body) => {
+  const jobAddingUrl = `${endpoint}/${jobId}:addOperations`;
+  const secondRequest = {
+    url: jobAddingUrl,
+    data: body.JSON,
+    headers,
+    method
+  };
+  const response = await httpSend(secondRequest);
+  return response;
+};
+
+/**
+ * This function helps to run a offlineUserDataJobs
+ * @param endpoint
+ * @param headers
+ * @param method
+ * @param jobId
+ */
+const runTheJob = async (endpoint, headers, method, jobId) => {
+  const jobRunningUrl = `${endpoint}/${jobId}:run`;
+  const thirdRequest = {
+    url: jobRunningUrl,
+
+    headers,
+    method
+  };
+  const response = await httpSend(thirdRequest);
+  return response;
+};
+
+/**
+ * This function is responsible for making the three steps required for uploding
+ * data to customer list.
+ * @param {*} request
+ * @returns
+ */
+const gaAudienceProxyRequest = async request => {
+  const { body, method, params, endpoint } = request;
+  const { headers } = request;
+  const { customerId, listId } = params;
+
+  // step1: offlineUserDataJobs creation
+
+  const firstResponse = await createJob(
+    endpoint,
+    customerId,
+    listId,
+    headers,
+    method
+  );
+  if (!isHttpStatusSuccess(firstResponse.response.status)) return firstResponse;
+
+  // step2: putting users into the job
+  const jobId = firstResponse.response.data.resourceName.split("/")[3];
+  const secondResponse = await addUserToJob(
+    endpoint,
+    headers,
+    method,
+    jobId,
+    body
+  );
+  if (!isHttpStatusSuccess(secondResponse.response.status))
+    return secondResponse;
+
+  // step3: running the job
+  const thirdResponse = await runTheJob(endpoint, headers, method, jobId);
+  return thirdResponse;
+};
+
+/**
+ * This function helps to detarmine type of error occured. According to the response
+ * we set authErrorCategory to take decision if we need to refresh the access_token
+ * or need to disable the destination.
+ * @param {*} code
+ * @param {*} response
+ * @returns
+ */
+const getAuthErrCategory = (code, response) => {
+  switch (code) {
+    case 401:
+      if (!response.error.details) return REFRESH_TOKEN;
+      return "";
+    case 403: // Access Denied
+      return DISABLE_DEST;
+    default:
+      return "";
+  }
+};
+
+const gaAudienceRespHandler = (destResponse, stageMsg) => {
+  const { status, response } = destResponse;
+  // const respAttributes = response["@attributes"] || null;
+  // const { stat, err_code: errorCode } = respAttributes;
+
+  if (isHttpStatusSuccess(status)) {
+    // Mostly any error will not have a status of 2xx
+    return response;
```

Comment:
We shouldn't return the destination response directly rather we should add a wrapper with our message that the request was processed successfully. 
`{
    status: status,
    message: "[Google adwords ..] : Request succeded with status ${status},
    destinationResponse: response
  }`

Reply:
We are calling this function from responseHandler where it is taken care of.